contributing-docs/testing/unit_tests.rst

```diff
@@ -0,0 +1,1160 @@
+
+.. code-block:: bash
+
+    breeze testing non-db-tests --backent postgres
+
+You can pass ``--parallel-test-type`` list of test types to execute or ``--exclude--parallel-test-types``
+to exclude them from the default set:.
+
+.. code-block:: bash
+
+    breeze testing db-tests --parallel-test-types "Providers API CLI"
+
+
+.. code-block:: bash
+
+    breeze testing db-tests --exclude-parallel-test-types "Providers API CLI"
+
+You can also run the same commands via ``breeze testing tests`` - by adding the necessary flags manually:
+
+.. code-block:: bash
+
+    breeze testing tests --run-db-tests-only --backend postgres --run-tests-in-parallel
+
+
+Also - if you want to iterate with the tests you can enter interactive shell and run the tests iteratively -
+either by package/module/test or by test type - whatever ``pytest`` supports.
+
+.. code-block:: bash
+
+    breeze shell --backend postgres --python 3.8
+    > pytest tests --run-db-tests-only
+
+As explained before, you cannot run DB tests in parallel using ``pytest-xdist`` plugin, but ``breeze`` has
+support to split all the tests into test-types to run in separate containers and with separate databases
+and you can run the tests using ``--run-tests-in-parallel`` flag (which is automatically enabled when
+you use ``breeze testing db-tests`` command):
+
+.. code-block:: bash
+
+    breeze testing tests --run-db-tests-only --backend postgres --python 3.8 --run-tests-in-parallel
+
+
+Best practices for DB tests
+...........................
+
+Usually when you add new tests you add tests "similar" to the ones that are already there. In most cases,
+therefore you do not have to worry about the test type - it will be automatically selected for you by the
+fact that the Test Class that you add the tests or the whole module will be marked with ``db_test`` marker.
+
+You should strive to write "pure" non-db unit tests (i.e. DB tests) but sometimes it's just better to plug-in
+the existing framework of DagRuns, Dags, Connections and Variables to use the Database directly rather
+than having to mock the DB access for example. It's up to you to decide.
+
+However, if you choose to write DB tests you have to make sure you add the ``db_test`` marker - either to
+the test method, class (with decorator) or whole module (with pytestmark at the top level of the module).
+
+In most cases when you add tests to existing modules or classes, you follow similar tests so you do not
+have to do anything, but in some cases you need to decide if your test should be marked as DB test or
+whether it should be changed to not use the database at all.
+
+If your test accesses the database but is not marked properly the Non-DB test in CI will fail with this message:
+
+.. code ::
+
+    "Your test accessed the DB but `_AIRFLOW_SKIP_DB_TESTS` is set.
+    Either make sure your test does not use database or mark your test with `@pytest.mark.db_test`.
+
+Marking test as DB test
+.......................
+
```

Comment:
Here as well, this chapter smells redundant to the one above?

Reply:
I moved it above, and renamed it to:

```
Examples of marking test as DB test
```